cas-server-support-spnego/src/main/java/org/jasig/cas/support/spnego/authentication/principal/SpnegoCredential.java

```diff
@@ -152,7 +152,11 @@ public boolean equals(final Object obj) {
 
     @Override
     public int hashCode() {
-        return Arrays.hashCode(this.getInitToken()) ^ Arrays.hashCode(this.getNextToken()) ^ this.principal.hashCode();
+        int hash = 3;
+        if (this.principal != null) {
+            hash = this.principal.hashCode();
+        }
+        return Arrays.hashCode(this.getInitToken()) ^ Arrays.hashCode(this.getNextToken()) ^ hash;
```

Comment:
Could we use a HashCodeBuilder, appending these arrays, and rather than 3 (which I think will make the build fail because it will be considered as a magic number), why don't init the hash with super.hashCode()?